### DISCUSSION TITLE
Is there type defination for `window. __TAURI__` object?

### DISCUSSION THREAD

**Discussion by percy507** [tauri-apps/tauri#3853] [Q&A]
I can't find it in tauri's source code.

**Comment by peterroe** [comment]
Hello, it will be import automatically，if you are using `@tauri-apps/api` ,

Such as : https://tauri.studio/docs/api/js/modules/app

```shell
Module: app
Get application metadata.

This package is also accessible with window.__TAURI__.app when tauri.conf.json > build > withGlobalTauri is set to true.
```

**Reply by percy507** [reply]
@peterroe  I mean the type of `window.__TAURI__` object, not the object itself.

```ts
// its type definition may like this
import type * as app from '@tauri-apps/api/app';

declare global {
  interface Window {
    __TAURI__: {
      app?: typeof app;
      // ... the other tauri modules
    };
  }
}
```

**Reply by dserv-nh** [reply]
Is there a way to import all modules at once? Previously that worked for me but after upgrading from 1.4.0 it no longer works and errors with `Property 'invoke' does not exist on type '{ convertFileSrc: (src: string, protocol: string) => string; }'`
```ts
import type TauriAPI from '@tauri-apps/api';
declare global {
    interface Window {
      __TAURI__?: typeof TauriAPI;
    }
}
```

<details><summary>More context</summary>
<p>

I use it inside a Vite app where it's located under `src/vite-env.d.ts`
I have some features that are unavailable when the app isn't run via Tauri so I use `window.__TAURI__` to check for that.
All Tauri API usages use `window.__TAURI__` too.

Full contents of `vite-env.d.ts`:

```ts
/// <reference types="vite/client" />

declare module '*.vue' {
  import type { DefineComponent } from 'vue';
  import type TauriAPI from '@tauri-apps/api';
  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/ban-types
  const component: DefineComponent<{}, {}, any>;
  export default component;

  global {
    interface Window {
      __TAURI__?: typeof TauriAPI;
    }
  }
}

declare const __APP_VERSION__: string;
```


</p>
</details>

**Reply by FreakeyPlays** [reply]
I started a new discussion regarding the same topic, but maybe one of you has the answer. How do you handle the types of Tauri plugins (like `os`, `dialog`, ...) in the window object?
